code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -1 +1,208 @@
+// ***************************************
+/datum/action/xeno_action/activable/mind_wrack
+	name = "Mind Wrack"
+	action_icon_state = "screech"
+	mechanics_text = "Afflicts the target's mind with disorienting and painful psychic energy. Effect is greater the closer the target is to your core, and the more xenos that are within 5 tiles when used. Must have a core or other xeno within 5 tiles to use."
+	ability_name = "mind wrack"
+	plasma_cost = 100
+	keybind_signal = COMSIG_XENOABILITY_MIND_WRACK
+	cooldown_timer = 30 SECONDS
+
+/datum/action/xeno_action/activable/mind_wrack/on_cooldown_finish()
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to assault the minds of our enemies.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/mind_wrack/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(QDELETED(target))
+		return FALSE
+
+	var/mob/living/carbon/C = target
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+
+	var/distance = get_dist(owner, C)
+
+	if(!iscarbon(target)) //only items and mobs can be flung.
+		to_chat(X, "<span class='xenowarning'>Our mind cannot interface with such an entity!</spam>")
+		return FALSE
+
+	if(isxeno(C))
+		var/mob/living/carbon/xenomorph/alien = C
+		if(alien.hivenumber == X.hivenumber)
+			to_chat(X, "<span class='xenonotice'>We would not assail the minds of our sisters!</span>")
+			return FALSE
+
+	if(distance > HIVEMIND_MIND_WRACK_MAX_RANGE)
+		to_chat(X, "<span class='xenowarning'>They are too far for us to reach their minds! The target must be [distance - HIVEMIND_MIND_WRACK_MAX_RANGE] tiles closer!</spam>")
+
+	if(!owner.line_of_sight(target, HIVEMIND_MIND_WRACK_MAX_RANGE))
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We can't focus our psychic energy properly without a clear line of sight!</span>")
+		return FALSE
+
+	if(!CHECK_BITFIELD(use_state_flags|override_flags, XACT_IGNORE_DEAD_TARGET) && C.stat == DEAD)
+		to_chat(X, "<span class='xenonotice'>Even we cannot touch the minds of the dead...</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/activable/mind_wrack/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+	var/mob/living/carbon/victim = A
+
+	var/power_level //What does the scouter say about it?
+	for(var/mob/living/carbon/xenomorph/ally in range(HIVEMIND_MIND_WRACK_POWER_CHORUS_RANGE, X ) ) //For each friendly xeno nearby, effect is more powerful
+		if(istype(ally) && ally.hivenumber == X.hivenumber)
+			++power_level //Increment the power level for each Xeno nearby
+		if(power_level > HIVEMIND_MIND_WRACK_POWER_MAX_CHORUS)
+			break
+
+	for(var/obj/effect/alien/hivemindcore/node_core in range(HIVEMIND_MIND_WRACK_POWER_CHORUS_RANGE, get_turf(X) ) ) //If our core is nearby, the effect is *especially* powerful.
+		if(node_core == X.core)
+			power_level = max(power_level, power_level + 2)
```

Comment:
A)use getdist instead here

B)Why are you doing a power_level = max(power_level, power_level +2)  just do power_level += 2

Reply:
Done.